test.js

```diff
@@ -419,11 +419,17 @@ test('deepKeys', t => {
 				a: 0,
 			},
 		},
+		'': {
+			a: 0,
+		},
 	}), [
-		'a\\.b.c.d',
+		'a\\.b.c.d[0]',
```

Comment:
If you want to be extra thorough you could cover the case where an array contains a complex object too, but I imagine it’s fine.